pkg/ebpf/tracee.go

```diff
@@ -185,12 +190,37 @@ type Tracee struct {
 	procInfo          *procinfo.ProcInfo
 	eventsSorter      *sorting.EventsChronologicalSorter
 	eventDerivations  map[int32]map[int32]deriveFn
+	kernelSymbols     *helpers.KernelSymbolTable
 }
 
 func (t *Tracee) Stats() *metrics.Stats {
 	return &t.stats
 }
 
+func isEssential(id int32) bool {
+	return EventsDefinitions[id].EssentialEvent
+}
+
+func setEssential(id int32) {
+	event := EventsDefinitions[id]
+	event.EssentialEvent = true
+	EventsDefinitions[id] = event
+}
+
+func handleEventsDependencies(e int32, initReq *RequiredInitValues) {
+	eDependencies := EventsDefinitions[e].Dependencies
+	if len(eDependencies.ksymbols) > 0 {
+		initReq.kallsyms = true
+	}
```

Comment:
Instead of initReq, you can populate a list with all the symbols required by the different events.
You can later use this list to initialize the bpf map